elastic/eui
# [Storybook] General consistency pass

## Summary

This PR is a general cleanup/consistency pass on our existing stories. The way we've written stories has evolved as we've learned more tips and tricks of Storybook's API, and this attempts to elevate all existing stories to the same level.

1. Adds new `hideStorybookControls()` and `disableStorybookControls()` utils so we're not copying the same `{ table: { disable: true } }` or `{ control: false }` over and over again
   - This has the main DRY benefit of future-proofing, if someday Storybook changes the API they use to render the controls table, we can quickly update it in one place instead of find+replacing hundreds of lines

2. Standardizes how and where we list component defaults - this should now always be in the initial component `meta.args` as opposed to defining them as separate `const`s and spreading them repeatedly.
    - This has the added benefit of receiving automatic prop name/type checking as opposed to manual.

3. More opinionated cleanups of a handful of spicier components

As always, I recommend [following along with changes by commit](https://github.com/elastic/eui/pull/7245/commits), as the final diff touches a lot of files and I'm doing a few different separate clean up things here.

## QA

- Go to https://eui.elastic.co/pr_7245/storybook/ 
- Click around and confirm stories are generally working as expected with useful props tables

### General checklist

N/A, dev only
